app/controllers/agents_controller.rb

```diff
@@ -179,7 +179,9 @@ def complete
 
   # Sanitize params[:return] to prevent open redirect attacks, a common security issue.
   def redirect_back(message)
-    if params[:return] == "show" && @agent
+    if params[:return] == "show" && @agent.destroyed?
```

Comment:
@dsander I did not understand exactly what you want. Because My English is not good.